crml/erc20-peg/src/lib.rs

```diff
@@ -261,47 +263,7 @@ decl_module! {
 		#[transactional]
 		pub fn withdraw(origin, asset_id: AssetId, amount: Balance, beneficiary: EthAddress) {
 			let origin = ensure_signed(origin)?;
-			ensure!(Self::withdrawals_active(), Error::<T>::WithdrawalsPaused);
-
-			// there should be a known ERC20 address mapped for this asset
-			// otherwise there may be no liquidity on the Ethereum side of the peg
-			let token_address = Self::asset_to_erc20(asset_id);
-			ensure!(token_address.is_some(), Error::<T>::UnsupportedAsset);
-			let token_address = token_address.unwrap();
-
-			if asset_id == T::MultiCurrency::staking_currency() {
-				let _result = T::MultiCurrency::transfer(
-					&origin,
-					&T::PegPalletId::get().into_account(),
-					asset_id,
-					amount, // checked amount < u128 in `deposit_claim` qed.
-					ExistenceRequirement::KeepAlive,
-				)?;
-			} else {
-				let _imbalance = T::MultiCurrency::withdraw(
-					&origin,
-					asset_id,
-					amount,
-					WithdrawReasons::TRANSFER,
-					frame_support::traits::ExistenceRequirement::KeepAlive,
-				)?;
-			}
-
-			let message = WithdrawMessage {
-				token_address,
-				amount: amount.into(),
-				beneficiary,
-			};
-
-			let claim_delay: Option<(Balance, T::BlockNumber)> = Self::claim_delay(asset_id);
-			if let Some((min_amount, delay)) = claim_delay {
-				if min_amount <= amount {
-					Self::delay_claim(delay, PendingClaim::Withdrawal(message));
-					return Ok(());
-				}
-			};
-			// process withdrawal immediately
-			Self::process_withdrawal(message, asset_id);
+			let _ = Self::do_withdrawal(origin, asset_id, amount, beneficiary, WithdrawCallOrigin::Runtime)?;
```

Comment:
```suggestion
			Self::do_withdrawal(origin, asset_id, amount, beneficiary, WithdrawCallOrigin::Runtime)
```

Reply:
I had to leave the ?; at the end to absorb the Ok(u64) case as the extrinsic doesn't use this value, only the EVM